### PR TITLE
Chore/3

### LINKED ISSUE #3: 開發環境基礎建置
1.建立專案基礎開發環境, 供後續TDD實作

### PR DESCRIPTION
**Pull request by IamSaker**
## Why need this change? / Root cause / Bug & Solution Description: 
- Build a development environment for the Mahjong project.
- Resolves #3

## Changes made:
- Build Python environment and package management.
- 

## Test Scope / Change impact:
-

### CODE REVIEW

**Review by shmily7829 — APPROVED** [review]
Approved, 請1/2會議時再協助說明安裝了哪些package及用途

File: .github/workflows/code-quality.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+          python-version: ${{ matrix.python-version }}
+      
+      - name: Black format
+        uses: psf/black@stable
+        with:
+          options: "--check --verbose -l 100"
+          jupyter: true
+          version: "22.10.0"
+      
+      # Process linter
```
Comment (FuyaosGithub) [inline .github/workflows/code-quality.yml:31]: 1. linter 已經寫上去了，未來會用在哪個部分呢？
2. black format 套件是否要先驗證後ＰＲ才會過？

File: .github/workflows/code-quality.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
+name: Code Quality
+on: 
+  pull_request:
+    branches: ["main", "release/*"]
```
[the remark this answers is not in the document]
Comment (IamSaker) [reply .github/workflows/code-quality.yml:4]: (保留擴充性) 那是 release branch，意思是 `release/` prefix 開頭的 branch 都會跑 GitHub Action

File: CODEOWNERS [hunk 3]
```diff
@@ -0,0 +1 @@
+* @Game-as-a-Service/mahjong
```
[the remark this answers is not in the document]
Comment (IamSaker) [reply CODEOWNERS:1]: [CODEOWNERS file location](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners#codeowners-file-location)
> To use a CODEOWNERS file, create a new file called CODEOWNERS in the root, docs/, or .github/ directory of the repository, in the branch where you'd like to add the code owners.

摁...放在哪我都沒有意見。

File: .github/workflows/code-quality.yml (hunk 1 above)
Comment (IamSaker) [reply .github/workflows/code-quality.yml:31]: 1. 用來提升 Code Quality
2. GitHub Action 會去執行這個 `.yml` 下的動作，所以可以這麼理解

**Review by Ritabear — APPROVED** [review]

File: .github/workflows/code-quality.yml (hunk 1 above)
Comment (IamSaker) [reply .github/workflows/code-quality.yml:31]: 進階一點可以做 [AST Checker](https://pylint.pycqa.org/en/latest/development_guide/how_tos/custom_checkers.html)

**Pull request merged**